File: strings/reverse_string/reverse_string.py
class Solution:
    def reverseString(self, s):
        """
        :type s: str
        :rtype: str
        """
        length = len(s)-1
        result = [None]*len(s)
        for c in s:
            result[length] = c
            length -= 1
        return "".join(map(str, result))

File: strings/reverse_string/reverse_string.java
class Solution {
    public String reverseString(String s) {
        byte[] sByteArray = s.getBytes();
        int length = sByteArray.length;
        byte[] result = new byte[length];
        for (int i = 0; i < length; i++) {
            result[i] = sByteArray[length-i-1];
        }
        return new String(result);
    }
}
